### PR TITLE
Student - Investigate and fix failing E2E tests

### PR DESCRIPTION
**Pull request by ndrsszsz**
- Simplified login logic to not use previously signed in account
- Updated the dates in PlannerTests (also updated the assignment dates in test account)

All of the Student E2E tests are passing (Bitrise might need to retry some tests because of flakiness)

[Bitrise run](https://app.bitrise.io/build/3d490bec-4f0e-4d63-933e-041d57d3e772)

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Build failed, skipping coverage check</td>
    </tr>
  </tbody>
</table>



[MBL-16679](https://instructure.atlassian.net/browse/MBL-16679)

<details><summary>:x: Swift lint</summary>


    ❌ /Users/vagrant/git/Core/Core/API/APIMock.swift:22:22: Blanket Disable Command Violation: The disabled 'force_try' rule should be re-enabled before the end of the file (blanket_disable_command)
    /Users/vagrant/git/Core/Core/DocViewer/DocViewerViewController.swift:224:25: error: Blanket Disable Command Violation: The enabled 'function_parameter_count' rule was not disabled (blanket_disable_command)
    /Users/vagrant/git/TestsFoundation/TestsFoundation/Element/Element.swift:180:1: error: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    ❌ /Users/vagrant/git/TestsFoundation/TestsFoundation/Element/Element.swift:184:1: Trailing Whitespace Violation: Lines should not have trailing whitespace (trailing_whitespace)
    
    
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against ee324a8bf77d1178b2e1702d6f1be5de3b47971d
</p>

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

File: TestsFoundation/TestsFoundation/CoreUIElements/Profile/Profile.swift [hunk 1]
```diff
@@ -44,7 +44,8 @@ public enum Profile: String, ElementWrapper {
 
     public static func open() {
         // HACK: sometimes a11y stuff doesn't show up...
-        if !Dashboard.profileButton.exists() {
+        let profileButton = Dashboard.profileButton.waitToExist(shouldFail: false)
+        if !profileButton.exists {
```
Comment (vargaat) [inline TestsFoundation/TestsFoundation/CoreUIElements/Profile/Profile.swift:48]: I see that this `exists` property still uses the snapshot != nil check to determine if the Element is there. We could use `rawElement.exists` in Element.exists function as this build-in method was much more reliable than the snapshot method.

Comment (vargaat) [reply TestsFoundation/TestsFoundation/CoreUIElements/Profile/Profile.swift:48]: Yes, that was it. I started the Student E2E test suite on this branch to see if anything went seriously bad.

**Review by vargaat — APPROVED** [review]

**Review by szabinst — APPROVED** [review]
Please remove jira related text from the PR title. Other than that LGTM!

**Review by idevainstr — APPROVED** [review]

**Pull request merged**